Sources/MapboxMaps/Foundation/Camera/CameraAnimator.swift

```diff
@@ -1,20 +1,40 @@
 import UIKit
+import CoreLocation
 
-// MARK: CameraAnimator Class
-public class CameraAnimator: NSObject {
+@objc public protocol CameraAnimatorProtocol: AnyObject {
+
+    /// Stops the animation in its tracks and calls any provided completion
+    func stopAnimation()
+
+    /// The current state of the animation
+    var state: UIViewAnimatingState { get }
+
+    /// TODO: Move this to a non-public interface
+    func update()
```

Comment:
Let's discuss how we might be able to do this.

Reply:
Try using 2 protocols, 1 public, 1 internal

Reply:
This is done.